luigi/db_task_history.py

```diff
@@ -65,13 +65,8 @@ def _session(self, session=None):
             yield session
         else:
             session = self.session_factory()
-            try:
+            with session.transaction:
```

Comment:
SQLAlchemy's session management does magic to make sure that if the rollback fails you still get the original exception that caused the rollback.

Also it looks nicer.